src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,68 @@
+package duke.task;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Represents a Task in the Duke program.
+ */
+public abstract class Task implements Serializable {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Creates a Task with a specific description.
+     * @param description the description of the task
+     */
+    public Task(String description) {
+        this.description = description;
+        isDone = false;
+    }
+
+
+    /**
+     * Returns the description of this task.
+     * @return the description of this task
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Sets the done-ness of the task.
+     * @param isDone the done-ness to be applied on the task.
+     */
+    public void setDone(boolean isDone) {
+        this.isDone = isDone;
+    }
+
+    /**
+     * Returns the status icon of the task, to identify whether it has been done.
+     * @return Y if the task has been done, X otherwise
+     */
+    public String getStatusIcon() {
```

Comment:
Are the brackets required here?

Reply:
Good point. I'll remove them in the next commit.